controllers/topics/broadcastMessage.js

```diff
@@ -0,0 +1,117 @@
+/* 
+    Send message to all users who join a topic
+*/
+const momentTz = require('moment-timezone');
+const {v4: uuidv4} = require('uuid');
+const UsersFunction = require('../../databases/functions/users');
+const {followTopicQueue} = require('../../services/redis/queue');
+const {User, CommunityMessageFormat, UserTopic} = require('../../databases/models');
+const {Op} = require('sequelize');
+
+const sendMessageToQueue = async (user_id, topic_id, community_message_format_id, delay) => {
+  const data = {
+    user_id,
+    topic_id,
+    community_message_format_id
+  };
+  let currentTime = momentTz().tz('America/Los_Angeles');
+  let requiredTime = momentTz().tz('America/Los_Angeles').add(delay, 'minutes');
+
+  const diffTime = requiredTime.diff(currentTime, 'milliseconds');
+
+  const options = {
+    jobId: uuidv4(),
+    removeOnComplete: true,
+    delay: diffTime
+  };
+  try {
+    await followTopicQueue.add(data, options);
+  } catch (e) {
+    console.log('error', e);
+  }
+};
+
+const calculateDelay = (joinDate, delay) => {
+  let currentTime = momentTz().tz('America/Los_Angeles');
+  let requiredTime = momentTz(joinDate).tz('America/Los_Angeles');
+
+  let diffTime = requiredTime.diff(currentTime, 'days');
+  diffTime = Math.abs(diffTime + delay);
+  return diffTime == 0 ? 60 * 1000 : diffTime * 24 * 60 * 60 * 1000;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The calculation of delay time seems to be done in milliseconds. This can lead to confusion and potential bugs in the future. It would be better to use a more human-readable format like minutes or hours and convert to milliseconds only when necessary.
```diff
-  let diffTime = requiredTime.diff(currentTime, 'days');
-  diffTime = Math.abs(diffTime + delay);
-  return diffTime == 0 ? 60 * 1000 : diffTime * 24 * 60 * 60 * 1000;
+  let diffTime = requiredTime.diff(currentTime, 'hours');
+  diffTime = Math.abs(diffTime + delay * 24); // assuming delay is in days
+  return diffTime == 0 ? 60 : diffTime * 60; // return delay in minutes
```
Then, convert this delay to milliseconds in `sendMessageToQueue` function.



<!-- This is an auto-generated comment by OSS CodeRabbit -->